inst/include/population_dynamics/fleet/fleet_nll.hpp

```diff
@@ -0,0 +1,128 @@
+/*! \file fleet_nll.hpp
+ *
+ * This File is part of the NOAA, National Marine Fisheries Service
+ * Fisheries Integrated Modeling System project. See LICENSE in the
+ * source folder for reuse information.
+ *
+ * Fleet nll file
+ * The purpose of this file is to serve as the class for fleet negative
+ * log likelihood.
+ *
+ * DEFINE guards for fleet nll outline to define the
+ * fleet hpp file if not already defined.
+ */
+#ifndef FIMS_POPULATION_DYNAMICS_FLEET_NLL_HPP
+#define FIMS_POPULATION_DYNAMICS_FLEET_NLL_HPP
+
+#include "../../common/model_object.hpp"
+#include "../../distributions/distributions.hpp"
+#include "fleet.hpp"
+#include "../../common/fims_math.hpp"
+#include "../../interface/interface.hpp"
+#include <numeric>
+
+namespace fims {
+#ifdef TMB_MODEL
+
+/** @brief fleet negative log-likelihood (nll).
+ *
+ * @tparam Type The type of the fleet functor.
+ *
+ */
+template <class Type>
+struct FleetIndexNLL : public Fleet<Type> {
+  static uint32_t id_g; /*!< reference id for fleet nll object*/
+
+  /** @brief Constructor.
+   */
+  FleetIndexNLL() { this->id = FleetIndexNLL::id_g++; }
+
+  virtual ~FleetIndexNLL() {}
+
+  /** @brief likelihood component function.
+   * Returns the negative log likelihood (nll).
+   * Based on equation (A.3.10) in Methot and Wetzel (2013)
+   * but with the addition of the constant terms.
+   */
+  Type evaluate() {
+    Type nll = 0.0; /*!< The negative log likelihood value */
+
+      fims::Dnorm<Type> dnorm;
```

Comment:
After M1, we might want to consider updating this to Dlnorm or provide flexibility to choose between the two

Reply:
will open a new issue in M2